### PR TITLE
feat: override publish button

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Overrides the Gutenberg publish button. The approach uses React Portals. The `Editor` component creates a DOM element and inserts in front of the publish button. This element is used as a portal to be filled with the custom publish button. The default publish button is hidden by CSS. The custom publish button's label reflects its purpose: "Send Newsletter" | "Sending Newsletter" | "Sent". The button is not enabled until the newsletter is ready to send, and consequently the pre-publish checklist has been removed. Currently there is no support for scheduling publishing. 

### How to test the changes in this Pull Request:

1. Create a Newsletter.
2. Verify publish button has text "Send Newsletter" and is disabled.
3. Select a list, input a sender name and email. Verify the publish button is now enabled.
4. Send the Newsletter. Verify the publish button is now disabled and reads "Sent."

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/components/send-button/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	const isButtonEnabled =
+		isPublishable &&
+		isSaveable &&
+		validationErrors &&
+		! validationErrors.length &&
+		'publish' !== status;
+	let label;
+	if ( 'publish' === status ) {
+		label = isSaving
+			? __( 'Sending Newsletter', 'newspack-newsletters' )
```
Comment (thomasguillot) [inline src/components/send-button/index.js:40]: To keep consistency with Posts and Pages. I suggest we change this to "Sending..."

File: src/components/send-button/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+			: __( 'Sent', 'newspack-newsletters' );
+	} else {
+		label = __( 'Send Newsletter', 'newspack-newsletters' );
```
Comment (thomasguillot) [inline src/components/send-button/index.js:43]: Same here. "Send"

**Comment by adekbadek** [comment]
I took at stab at supporting post scheduling in 4f0addf – let me know if it makes sense.

File: src/components/send-button/index.js (hunk 2 above)
Comment (jeffersonrabb) [reply src/components/send-button/index.js]: Fixed in https://github.com/Automattic/newspack-newsletters/pull/123/commits/98ad8765ad45fafc60a693cfd0f7ca716d58125e

File: src/components/send-button/index.js (hunk 1 above)
Comment (jeffersonrabb) [reply src/components/send-button/index.js]: Fixed in https://github.com/Automattic/newspack-newsletters/pull/123/commits/98ad8765ad45fafc60a693cfd0f7ca716d58125e

**Comment by jeffersonrabb** [comment]
> I took at stab at supporting post scheduling in 4f0addf – let me know if it makes sense.

Looks good to me! If you haven't already could you eyeball it next to https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/post-publish-button/index.js and verify the logic is similar?

**Comment by adekbadek** [comment]
Yeah, it looks similar

**Comment by adekbadek** [comment]
I only wonder if and how we should handle [`pending` status](https://wordpress.org/support/article/post-status/#pending) – ?

**Comment by jeffersonrabb** [comment]
> I only wonder if and how we should handle [`pending` status](https://wordpress.org/support/article/post-status/#pending) – ?

In https://github.com/Automattic/newspack-newsletters/pull/123/commits/db15bc67fcc001d6dd3034e968122b2d82bb0df7 I copied the handling of `pending` directly from https://github.com/WordPress/gutenberg/blob/master/packages/editor/src/components/post-publish-button/index.js

File: src/editor/sidebar/index.js [hunk 3]
```diff
@@ -209,7 +209,10 @@ class Sidebar extends Component {
 					<Button
 						isLink
 						onClick={ () => this.updateSender( senderName, senderEmail ) }
-						disabled={ inFlight || ( senderEmail.length ? ! hasValidEmail( senderEmail ) : false ) }
+						disabled={
+							inFlight ||
+							( senderEmail && senderEmail.length ? ! hasValidEmail( senderEmail ) : false )
+						}
```
Comment (adekbadek) [inline src/editor/sidebar/index.js:215]: That won't be needed, #119 fixed it

Comment (jeffersonrabb) [reply src/editor/sidebar/index.js]: Removed in https://github.com/Automattic/newspack-newsletters/pull/123/commits/ce6d7676b4408dce9c376dc21ff61a3da2c5b5d9

**Pull request merged**